diff -ur ogre/acinclude.m4 ogre.new/acinclude.m4
--- ogre/acinclude.m4	2008-08-08 12:58:28.000000000 +0400
+++ ogre.new/acinclude.m4	2008-10-02 23:29:58.000000000 +0400
@@ -305,19 +305,7 @@
               [build_exr=$enableval],
               [build_exr=no])
 
-if test "x$build_exr" = "xyes" ; then
-	PKG_CHECK_MODULES(OPENEXR, OpenEXR, [build_exr=yes], [build_exr=no])
-
-	if test "x$build_exr" = "xyes" ; then
-	   	AC_CONFIG_FILES([ PlugIns/EXRCodec/Makefile \
-    					 PlugIns/EXRCodec/src/Makefile \
-    					 PlugIns/EXRCodec/include/Makefile])
-		AC_SUBST(OPENEXR_CFLAGS)
-		AC_SUBST(OPENEXR_LIBS)
-
-	fi
-
-fi
+build_exr="no"
 
 AM_CONDITIONAL(BUILD_EXRPLUGIN, test x$build_exr = xyes)
 
@@ -446,19 +434,7 @@
 AC_DEFUN([OGRE_CHECK_CEGUI], [
     PKG_CHECK_MODULES(CEGUI, CEGUI >= 0.5.0, 
             [build_cegui_sample=true], [build_cegui_sample=false])
-    if test x$build_cegui_sample = xtrue; then
-        AC_CONFIG_FILES([Samples/Common/CEGUIRenderer/Makefile \
-                         Samples/Common/CEGUIRenderer/CEGUI-OGRE.pc
-                         Samples/Common/CEGUIRenderer/src/Makefile \
-                         Samples/Common/CEGUIRenderer/include/Makefile \
-                         Samples/Gui/Makefile \
-                         Samples/Gui/src/Makefile])
-        AC_SUBST(CEGUI_CFLAGS)
-        AC_SUBST(CEGUI_LIBS)
-        AC_MSG_RESULT([CEGUI available, Gui and FacialAnimation samples will be built])
-    else
-        AC_MSG_RESULT([CEGUI not available, Gui and FacialAnimation samples will not be built])
-    fi
+	build_cegui_sample="false"
     AM_CONDITIONAL([HAVE_CEGUI], [test x$build_cegui_sample = xtrue])
 ])
 
diff -ur ogre/configure.in ogre.new/configure.in
--- ogre/configure.in	2008-06-12 18:10:57.000000000 +0400
+++ ogre.new/configure.in	2008-10-02 23:31:22.000000000 +0400
@@ -78,9 +78,13 @@
 AC_SUBST(ZZIPLIB_CFLAGS)
 AC_SUBST(ZZIPLIB_LIBS)
 
+PKG_CHECK_MODULES(OGRE, OGRE)
+CXXFLAGS="$CXXFLAGS $OGRE_CFLAGS"
+LDFLAGS="$LDFLAGS $OGRE_LIBS"
+
 dnl Get the GL and Platform libs for target build
 OGRE_GET_PLATFORM
-OGRE_GET_GLSUPPORT
+dnl OGRE_GET_GLSUPPORT
 
 dnl Find the optional components we should build
 OGRE_CHECK_GUI
@@ -88,7 +92,9 @@
 OGRE_CHECK_THREADING
 OGRE_CHECK_FREEIMAGE
 OGRE_CHECK_DEVIL
+AC_LANG_PUSH(C++)
 OGRE_CHECK_CG
+AC_LANG_POP
 OGRE_CHECK_OPENEXR
 OGRE_CHECK_DX9
 OGRE_CHECK_SSE
@@ -120,137 +126,10 @@
 
 dnl Create the makefiles
 AC_CONFIG_FILES([Makefile \
-    OgreMain/Makefile \
-    OgreMain/src/Makefile \
-    OgreMain/include/Makefile \
-    OgreMain/include/GLX/Makefile \
-    OgreMain/include/gtk/Makefile \
     PlugIns/Makefile \
-    PlugIns/BSPSceneManager/Makefile \
-    PlugIns/BSPSceneManager/src/Makefile \
-    PlugIns/BSPSceneManager/include/Makefile \
     PlugIns/CgProgramManager/Makefile \
     PlugIns/CgProgramManager/src/Makefile \
-    PlugIns/CgProgramManager/include/Makefile \
-    PlugIns/OctreeSceneManager/Makefile \
-    PlugIns/OctreeSceneManager/src/Makefile \
-    PlugIns/OctreeSceneManager/include/Makefile \
-    PlugIns/ParticleFX/Makefile \
-    PlugIns/ParticleFX/src/Makefile \
-    PlugIns/ParticleFX/include/Makefile \
-    RenderSystems/Makefile \
-    RenderSystems/GL/Makefile \
-    RenderSystems/GL/src/Makefile \
-    RenderSystems/GL/include/Makefile \
-    RenderSystems/GL/src/atifs/Makefile \
-    RenderSystems/GL/src/atifs/include/Makefile \
-    RenderSystems/GL/src/atifs/src/Makefile \
-    RenderSystems/GL/src/GLSL/Makefile \
-    RenderSystems/GL/src/GLSL/include/Makefile \
-    RenderSystems/GL/src/GLSL/src/Makefile \
-    RenderSystems/GL/src/nvparse/Makefile \
-    RenderSystems/Direct3D9/Makefile \
-    RenderSystems/Direct3D9/src/Makefile \
-    RenderSystems/Direct3D9/include/Makefile \
-    Samples/Makefile \
-    Samples/Common/Makefile \
-    Samples/Common/include/Makefile \
-    Samples/Common/bin/Makefile \
-    Samples/BezierPatch/Makefile \
-    Samples/BezierPatch/src/Makefile \
-    Samples/BezierPatch/include/Makefile \
-    Samples/CameraTrack/Makefile \
-    Samples/CameraTrack/src/Makefile \
-    Samples/CelShading/Makefile \
-    Samples/CelShading/src/Makefile \
-    Samples/Compositor/Makefile \
-    Samples/Compositor/src/Makefile \
-    Samples/Compositor/include/Makefile \
-    Samples/CubeMapping/Makefile \
-    Samples/CubeMapping/src/Makefile \
-    Samples/CubeMapping/include/Makefile \
-    Samples/DeferredShading/Makefile \
-    Samples/DeferredShading/src/Makefile \
-    Samples/DeferredShading/include/Makefile  \
-    Samples/Dot3Bump/Makefile \
-    Samples/Dot3Bump/src/Makefile \
-    Samples/EnvMapping/Makefile \
-    Samples/EnvMapping/src/Makefile \
-    Samples/EnvMapping/include/Makefile \
-    Samples/FacialAnimation/Makefile \
-    Samples/FacialAnimation/src/Makefile \
-    Samples/Fresnel/Makefile \
-    Samples/Fresnel/src/Makefile \
-    Samples/Grass/Makefile \
-    Samples/Grass/src/Makefile \
-    Samples/Transpacency/Makefile \
-    Samples/Transpacency/src/Makefile \
-    Samples/Transpacency/include/Makefile \
-    Samples/Lighting/Makefile \
-    Samples/Lighting/src/Makefile \
-    Samples/Lighting/include/Makefile \
-    Samples/OceanDemo/Makefile \
-    Samples/OceanDemo/src/Makefile \
-    Samples/OceanDemo/include/Makefile \
-    Samples/ParticleFX/Makefile \
-    Samples/ParticleFX/src/Makefile \
-    Samples/ParticleFX/include/Makefile \
-    Samples/RenderToTexture/Makefile \
-    Samples/RenderToTexture/src/Makefile \
-    Samples/TextureFX/Makefile \
-    Samples/TextureFX/src/Makefile \
-    Samples/TextureFX/include/Makefile \
-    Samples/Shadows/Makefile \
-    Samples/Shadows/src/Makefile \
-    Samples/SkyBox/Makefile \
-    Samples/SkyBox/src/Makefile \
-    Samples/SkyBox/include/Makefile \
-    Samples/SkyDome/Makefile \
-    Samples/SkyDome/src/Makefile \
-    Samples/SkyDome/include/Makefile \
-    Samples/SkyPlane/Makefile \
-    Samples/SkyPlane/src/Makefile \
-    Samples/SkyPlane/include/Makefile \
-    Samples/Smoke/Makefile \
-    Samples/Smoke/src/Makefile \
-    Samples/Smoke/include/Makefile \
-    Samples/BSP/Makefile \
-    Samples/BSP/src/Makefile \
-    Samples/SkeletalAnimation/Makefile \
-    Samples/SkeletalAnimation/src/Makefile \
-    Samples/SkeletalAnimation/include/Makefile \
-    Samples/Terrain/Makefile \
-    Samples/Terrain/src/Makefile \
-    Samples/Terrain/include/Makefile \
-    Samples/Water/Makefile \
-    Samples/Water/src/Makefile \
-    Samples/Water/include/Makefile \
-    Samples/DynTex/Makefile \
-    Samples/DynTex/src/Makefile \
-    Samples/DynTex/include/Makefile \
-    Samples/VolumeTex/Makefile \
-    Samples/VolumeTex/src/Makefile \
-    Samples/VolumeTex/include/Makefile \
-    Samples/Instancing/Makefile \
-    Samples/Instancing/src/Makefile \
-    Samples/Instancing/include/Makefile \
-    Tests/Makefile \
-    Tests/src/Makefile \
-    Tools/Makefile \
-    Tools/MaterialUpgrader/Makefile \
-    Tools/MaterialUpgrader/src/Makefile \
-    Tools/MaterialUpgrader/include/Makefile \
-    Tools/MeshUpgrader/Makefile \
-    Tools/MeshUpgrader/src/Makefile \
-    Tools/XMLConverter/Makefile \
-    Tools/XMLConverter/src/Makefile \
-    Tools/XMLConverter/include/Makefile \
-    Tools/rcapsdump/Makefile \
-    Tools/rcapsdump/src/Makefile \
-    Tools/LightwaveConverter/Makefile \
-    Tools/LightwaveConverter/src/Makefile \
-    Tools/LightwaveConverter/include/Makefile \
-    OGRE.pc
+    PlugIns/CgProgramManager/include/Makefile
     ])
 
 AC_OUTPUT
diff -ur ogre/Makefile.am ogre.new/Makefile.am
--- ogre/Makefile.am	2008-06-12 18:10:57.000000000 +0400
+++ ogre.new/Makefile.am	2008-10-02 23:29:58.000000000 +0400
@@ -1,6 +1,3 @@
-SUBDIRS = OgreMain PlugIns RenderSystems Samples Tools Tests
+SUBDIRS = PlugIns
 ACLOCAL_AMFLAGS = -I Scripts/m4
 AUTOMAKE_OPTIONS = foreign
-
-pkgconfigdir = $(libdir)/pkgconfig
-pkgconfig_DATA = OGRE.pc
diff -ur ogre/PlugIns/Makefile.am ogre.new/PlugIns/Makefile.am
--- ogre/PlugIns/Makefile.am	2008-06-12 18:10:57.000000000 +0400
+++ ogre.new/PlugIns/Makefile.am	2008-10-02 23:29:58.000000000 +0400
@@ -1,9 +1,5 @@
-SUBDIRS = BSPSceneManager ParticleFX OctreeSceneManager 
+SUBDIRS =
 
 if BUILD_CGPLUGIN
 SUBDIRS += CgProgramManager
 endif
-
-if BUILD_EXRPLUGIN
-SUBDIRS += EXRCodec
-endif
